### PR TITLE
Descriptors

### PR DESCRIPTION
**Pull request by rpautrat**

### CODE REVIEW

File: superpoint/models/super_point.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+                              [batch_size, 1])  # shape = (n_batches x Hc x Wc, 2)
+        coord_cells = tf.reshape(coord_cells, [batch_size, Hc, Wc, 2])
+        warped_coord_cells = tf.reshape(warped_coord_cells, [batch_size, Hc, Wc, 2])
+        coord_cells = tf.cast(tf_repeat(coord_cells, [1, Hc, Wc, 1]), tf.float32)
+        warped_coord_cells = tf.cast(tf.tile(warped_coord_cells, [1, Hc, Wc, 1]),
+                                     tf.float32)
+        # shape = (n_batches, Hc x Hc, Wc x Wc, 2)
+
+        # Compute the pairwise distances and filter the distances less than the threshold
+        cell_distances = tf.norm(coord_cells - warped_coord_cells, axis=-1)
```
Comment (Skydes) [inline superpoint/models/super_point.py]: My guess is that `coord_cells` should be rescaled by 0.5 or the homography should be tweaked.

File: superpoint/models/super_point.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+                                  tf.ones([Hc * Wc], dtype=tf.float32),
+                                  full_shape)  # shape = (H, W)
+        map_cells = tf.tile(tf.expand_dims(map_cells, 0),
+                            [batch_size, 1, 1])  # shape = (n_batches, H, W)
+
+        # Compute the position of the warped center pixels
+        warped_coord_cells = tf.map_fn(warp_keypoints_to_list,
+                                       (map_cells, H),
+                                       dtype=tf.float32)
+        # shape = (n_batches x Hc x Wc, 2)
```
Comment (Skydes) [inline superpoint/models/super_point.py]: Shouldn't that be `(n_batches, Hc x Wc, 2)`?

File: superpoint/models/super_point.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        # Switch to 'channels last' once and for all
+        if config['data_format'] == 'channels_first':
+            logits = tf.transpose(logits, [0, 2, 3, 1])
+            warped_logits = tf.transpose(warped_logits, [0, 2, 3, 1])
+            descriptors = tf.transpose(descriptors, [0, 2, 3, 1])
+            warped_descriptors = tf.transpose(warped_descriptors, [0, 2, 3, 1])
+
+        # Resize the warped data to match the shape of the original data
+        shape = tf.shape(logits)[1:3]
+        warped_logits = tf.image.resize_images(warped_logits, shape)
```
Comment (Skydes) [inline superpoint/models/super_point.py]: Wouldn't it be cleaner to instead rescale the homography fed to `warp_keypoints_to_map`?

Comment (rpautrat) [reply superpoint/models/super_point.py]: The issue is that the output of `warp_keypoints_to_map` is of the same shape as its input keypoint map, i.e. `inputs['keypoint_map']`, but we would like a ground truth twice as small if we don't resize `warped_logits` as it is the case currently.
One quick fix would be to modify `warp_keypoints_to_map` and add an argument setting the output shape that we want. What do you think?

**Pull request merged**